backend/app/Models/OverallComment.php

```diff
@@ -42,7 +45,12 @@ public function submission(): BelongsTo
      */
     public function replies(): HasMany
     {
-        return $this->hasMany(OverallComment::class, 'parent_id');
+        $thread_replies = $this->hasMany(OverallComment::class, 'parent_id');
+        if ($thread_replies->count() > 0) {
+            return $thread_replies;
+        } else {
+            return $this->hasMany(OverallComment::class, 'reply_to_id');
+        }
```

Comment:
I think (at a cursory glance) that these functions and the attribute functions are the same for Inline and Overall comment models.  You may want to consider implementing them as a trait (`use DeleteableComments;` ? ).

You'll need to use `__CLASS__` or `get_class()` instead of `OverallComment::class` or `InlineComment::class` though.

Reply:
I'll see what I can do